### DISCUSSION TITLE
Setting JAVA_HOME to brewed 'eclipse-java' java location?

### DISCUSSION THREAD

**Discussion by mgd** [Homebrew/discussions#1145] [Everyday usage]
Not really sure this is a 'home-brew' issue per say, but hoping someone can point me in the right direction...
I am running into an issue where I cannot set my env var JAVA_HOME to the java version (15) brewed from 'eclipse-java'.
The JDK was installed to `/usr/local/opt/java/libexec/openjdk.jdk`
Setting the variable via command line `export JAVA_HOME='/usr/local/opt/java/libexec'` does not work, as I get
`% /usr/libexec/java_home -V
The operation couldn’t be completed. Unable to locate a Java Runtime.
Please visit http://www.java.com for information on installing Java.`

Now I am on Big Sur, which has issues with JAVA_HOME never being set _initially_
https://stackoverflow.com/questions/15624667/mac-osx-java-terminal-version-incorrect
https://developer.apple.com/forums/thread/666681
However, after reading through, I was under the impression what I am doing would correct this. Similarly I have tried the export in my .zshrc profile, closed out of the terminal application and restarted but no such luck there either. Any help is appreciated!

**Comment by carlocab** [comment]
I think it should be something like
```
export JAVA_HOME='/usr/local/opt/java/libexec/openjdk.jdk/Contents/Home'
```
Of course, check that this path exists first before setting it. This directory should contain something that looks like
```
bin  conf  demo  include  jmods  legal  lib  man  release
```

**Reply by mgd** [reply]
That fixed it!
